Increase clarity around retired themes

dsas opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTTHEM-118 (#102498)

## Proposed Changes

* On the theme details screen don't show a theme badge for retired themes unless it is the users active theme. This reduces confusion where users may think that upgrading plan will give them access to the retired theme, while continuing to be clear that where it is the active theme, that it's use may be contingent on their current plan.
* On the theme details screen, do show preview + customize buttons for the active theme regardless of whether the theme is retired or not.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Improve clarity around what theme retirement means
* Reduce scope for confusion around plan upgrades and retired themes

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

There are three test scenarios for the theme showcase:
 1. The global themes view at /theme/:retired-theme-slug
 2. The site-specific calypso themes view for a site with a retired theme active (you can enable a retired theme via the cli/brc) /theme/:retired-theme-slug/:site-with-that-theme-active
 3. The site-specific calypso themes view for a site without a retired theme active: /theme/:retired-theme-slug/:site-without-that-theme-active

For each scenario, it should be clear that the theme is retired and what this means. Specifically, you should see a notice with a link and should not see anything that looks like an upsell for the theme.

If the theme is active you should now be able to see the preview and customise buttons. You should never be able to see other buttons.

You should also check the onboarding design picker (/setup/site-setup/designSetup?siteSlug=:site-slug) ( and the theme showcase grid (/themes) to check nothing has changed there.









Context | Before | After
-------|------|------
Global | <img width="1326" height="969" alt="Screenshot 2025-11-13 at 20 23 20" src="https://github.com/user-attachments/assets/6667060d-1983-47c6-b8dc-f46a2c5ec9b2" /> | <img width="1323" height="969" alt="Screenshot 2025-11-17 at 14 31 57" src="https://github.com/user-attachments/assets/b0f7b21f-4270-4aed-9aa0-9839477fc723" /> 
Site w/o theme | <img width="1326" height="969" alt="Screenshot 2025-11-13 at 20 23 37" src="https://github.com/user-attachments/assets/aa682edc-0b43-4ee7-9c4c-cc14ca3cdbd5" /> | <img width="1326" height="966" alt="Screenshot 2025-11-13 at 22 26 11" src="https://github.com/user-attachments/assets/dfd00755-3bf1-497a-a1ec-af14bdff03e8" /> | <img width="1323" height="969" alt="Screenshot 2025-11-17 at 14 36 32" src="https://github.com/user-attachments/assets/e7b2745a-ba7f-4929-8fcb-e3521f155587" />
Site with theme | <img width="1326" height="969" alt="Screenshot 2025-11-13 at 20 23 51" src="https://github.com/user-attachments/assets/201511b9-b686-4469-99d5-abb467dbf907" />|  <img width="1323" height="969" alt="Screenshot 2025-11-17 at 14 37 43" src="https://github.com/user-attachments/assets/00167ee8-bef4-4ed7-a13b-15ca586e629f" />
LOTS | <img width="1323" height="969" alt="Screenshot 2025-11-17 at 14 26 25" src="https://github.com/user-attachments/assets/d5bc1729-ae8b-4405-8cff-7418ea1ac9d8" /> | <img width="1323" height="969" alt="Screenshot 2025-11-17 at 14 34 37" src="https://github.com/user-attachments/assets/03d41fdb-34dc-44bd-beba-86a47aa1d5e5" />

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

dsas committed:
Increase clarity around retired themes

On the theme details screen don't show a theme badge for retired themes
unless it is the users active theme. This reduces confusion where users
may think that upgrading plan will give them access to the retired
theme, while continuing to be clear that where it is the active theme,
that it's use may be contingent on their current plan.

dsas committed:
Fix prettier error

dsas committed:
Update the theme component for retired themes

Pass the new `isThemeRetired` and `isThemeActiveForSite` props to
ThemeTierBadge from the Theme component.

This is a 'belt and braces' change, as this component is currently only
used in theme grids, which only display retired themes if they're
active, in which case the active badge takes precedence over the badge
here anyway.

dsas committed:
Update design picker ThemeTierBadge usage

Pass the new `isThemeRetired` and `isThemeActiveForSite` props in the
onboarding design picker. The design picker can't include retired
themes - it isn't even aware of the concept, so a less thorough approach
has been taken here.

dsas committed:
Add retired notice to the theme sheet

This adds a prominent notice to the theme sheet explaining that the
theme is retired and what this means, with a link to the support centre
to learn more.

dsas committed:
Update snapshot

dsas committed:
Allow previewing retired yet active themes

dsas committed:
Show customize button on retired, yet active themes

mmtr reviewed (commented):
The notice is misaligned in a logged-out context, and the "Learn more" link does not work:

<img width="1728" height="782" alt="Image" src="https://github.com/user-attachments/assets/031c7a41-fb62-41d5-8eda-2070cbd03c12" />

dsas committed:
Improve alignment of retirement notice

The WordPress Notice component assumes it should be the full width of
the container. This doesn't look correct for the /themes page.

An earlier commit added alignment for the global logged-in showcase but
the logged-out showcase was left full width and the with-site pages were
not exactly lined up.

This change brings the same consist alignment from the global logged-in
showcase to all of the other versions of the theme showcase.

mmtr commented:
> The notice is misaligned in a logged-out context

Just realized that this happens on logged-in contexts as well, when the viewport is wide enough:

<img width="2141" height="906" alt="Screenshot 2025-11-17 at 12 22 04" src="https://github.com/user-attachments/assets/0274f024-b263-47ff-b3a1-62b80edccf75" />

dsas committed:
Simplify the retirement notice alignment by applying the navigation header rules

dsas committed:
Change InlineSupportLink behaviour depending on login status

The logged-out theme showcase doesn't support the help centre, so the
InlineSupportLink should behave like a boring old link there.
